########CAMERA MISSING SYMBOLS FOR LEECO 1S BY WISNIAPL ################
project frameworks/av/
diff --git a/camera/CameraParameters.cpp b/camera/CameraParameters.cpp
old mode 100644
new mode 100755
index 76f92e7..1cc278d
--- a/camera/CameraParameters.cpp
+++ b/camera/CameraParameters.cpp
@@ -131,6 +131,8 @@ const char CameraParameters::FLASH_MODE_AUTO[] = "auto";
 const char CameraParameters::FLASH_MODE_ON[] = "on";
 const char CameraParameters::FLASH_MODE_RED_EYE[] = "red-eye";
 const char CameraParameters::FLASH_MODE_TORCH[] = "torch";
+const char CameraParameters::FLASH_MODE_TORCH1[] = "torch1";
+const char CameraParameters::FLASH_MODE_TORCH2[] = "torch2";
 
 // Values for scene mode settings.
 const char CameraParameters::SCENE_MODE_AUTO[] = "auto";
diff --git a/include/camera/CameraParameters.h b/include/camera/CameraParameters.h
old mode 100644
new mode 100755
index 74517b4..1ea69e5
--- a/include/camera/CameraParameters.h
+++ b/include/camera/CameraParameters.h
@@ -587,6 +587,8 @@ public:
     // Constant emission of light during preview, auto-focus and snapshot.
     // This can also be used for video recording.
     static const char FLASH_MODE_TORCH[];
+    static const char FLASH_MODE_TORCH1[];
+    static const char FLASH_MODE_TORCH2[];
 
     // Values for scene mode settings.
     static const char SCENE_MODE_AUTO[];

diff --git a/camera/cameraserver/cameraserver.rc b/camera/cameraserver/cameraserver.rc
index fea5a1d..08277d6 100644
--- a/camera/cameraserver/cameraserver.rc
+++ b/camera/cameraserver/cameraserver.rc
@@ -1,6 +1,6 @@
 service cameraserver /system/bin/cameraserver
     class main
     user cameraserver
-    group audio camera input drmrpc
+    group audio camera input drmrpc media system
     ioprio rt 4
     writepid /dev/cpuset/camera-daemon/tasks /dev/stune/top-app/tasks
